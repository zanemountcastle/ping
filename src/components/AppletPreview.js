import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { Card } from 'react-native-elements';

export default class AppletPreview extends Component {

  render() {
    return (
      <Card title={this.props.data.title} >
        <Text>Ping me when</Text>
        <Text>{this.props.data.organization}</Text>
      </Card>
    );
  }
}
