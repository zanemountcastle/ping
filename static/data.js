export const feeds = [
  {
    "id": "111",
    "title": "There's food in the Baraha.",
    "organization": "NYUAD",
    "color": "#03C7F5",
    "status": "On",
    "lastActive": "one minute ago",
  },
  {
    "id": "112",
    "title": "A new issue of The Gazelle is released.",
    "organization": "The Gazelle",
    "color": "#A327FD",
    "status": "Off",
    "lastActive": "three days ago",
  },
  {
    "id": "113",
    "title": "The A6 hammock is free.",
    "organization": "ResEd",
    "color": "#628395",
    "status": "On",
    "lastActive": "four hours ago",
  },
  {
    "id": "114",
    "title": "The Saints score a touchdown.",
    "organization": "The Saints",
    "color": "#F0CF65",
    "status": "On",
    "lastActive": "one minute ago",
  },
  {
    "id": "115",
    "title": "My coffee is ready.",
    "organization": "Blacksmith Coffee",
    "color": "#40403F",
    "status": "On",
    "lastActive": "one minute ago",
  }
];
