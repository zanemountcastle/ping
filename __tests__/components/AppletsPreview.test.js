import React from 'react';
import AppletPreview from '../../src/components/AppletPreview';
import { feeds } from '../../static/data.js';

import renderer from 'react-test-renderer';

describe('applet preview', () => {

  it('renders without crashing', () => {
    const rendered = renderer.create(<AppletPreview applet={feeds[0]} />).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AppletPreview applet={feeds[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
