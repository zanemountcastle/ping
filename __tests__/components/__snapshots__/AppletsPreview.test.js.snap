// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applet preview renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#03C7F5",
      "borderRadius": 10,
      "flexDirection": "column",
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 20,
      "minHeight": 200,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 25,
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontSize": 16,
          "fontWeight": "700",
          "marginTop": 10,
        }
      }
    >
      Ping me when
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "height": 2,
          "marginTop": 7,
          "width": 35,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontSize": 24,
          "fontWeight": "700",
          "marginTop": 12,
        }
      }
    >
      There's food in the Baraha.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(255,255,255,0.8)",
          "fontSize": 12,
          "fontWeight": "700",
          "marginTop": 7,
        }
      }
    >
      by 
      NYUAD
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "bottom": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
        "paddingBottom": 5,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 5,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(255,255,255,0.5)",
          "fontWeight": "700",
        }
      }
    >
      On
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(255,255,255,0.5)",
          "fontWeight": "700",
        }
      }
    >
      one minute ago
    </Text>
  </View>
</View>
`;
